---
title: "EEBOB R Assignment Notebook - Benjamin Trampe"
output: html_notebook
---
Gets current working directory.
```{r}
getwd()

```
Create a folder called R_Assignment.
Download genotype files called "fang_et_al_genotypes.txt" and "snp_position.txt" and add them to the R_Assignment folder.
Setting working directory to the folder that was just created call R_Assignment.
```{r setup}
knitr::opts_knit$set(root.dir ="~/R_Assignment")
```
A function to install the require the R packages if not already present for the assignment.
```{r}
installingpackages <- function(x){
  for( i in x ){
    if( ! require( i , character.only = TRUE ) ){
      install.packages( i , dependencies = TRUE)
      require( i , character.only = TRUE )
    }
  }
}
installingpackages( c("tidyr","dplyr","ggplot2" , "reshape2") )
```
Loads all of the require packages. 
```{r}
library(ggplot2)
library(tidyr)
library(plyr);library(dplyr)
library(reshape2)
```
Loads the text files form the the data frame.
```{r}
genotype <- as.data.frame(read.table("~/R_Assignment/fang_et_al_genotypes.txt", sep="\t",header=TRUE))
snppos <- as.data.frame(read.table("~/R_Assignment/snp_position.txt", sep="\t",header=TRUE))
```
Gets the file size for each file
```{r}
file.info("~/R_Assignment/fang_et_al_genotypes.txt")$size
file.info("~/R_Assignment/snp_position.txt")$size
```
Gives the number of rows in each data frame.
```{r}
nrow(genotype)
nrow(snppos)
```
Gives the number of columns in each data frame.
```{r}
ncol(genotype)
ncol(snppos)
```
Gives the structure of the data frame. 
```{r}
str(genotype)
str(snppos)
```
Sub setting the data to get a data frame for maize and teosinte based on the required groups
```{r}
maize <- genotype[which(genotype$Group=="ZMMIL" | genotype$Group =="ZMMLR" | genotype$Group == "ZMMMR"),]
teosinte <-genotype[which(genotype$Group=="ZMPBA" | genotype$Group =="ZMPIL" | genotype$Group == "ZMPJA"),]
```
Views the files to verify the correct sub setting was completed
```{r}
View(maize)
View(teosinte)
```
Sub setting the SNP Position to leave the required data of SNP_ID, Chromosome, and Position. It changes all of the unknown chromosomes and positions to NA>
```{r}
snp_sub <- select(snppos,SNP_ID,Chromosome,Position) 
snp_sub$Chromosome[snp_sub$Chromosome == "unknown"] <- NA
snp_sub$Chromosome[snp_sub$Chromosome == "multiple"] <- NA
snp_sub$Position[snp_sub$Position == "unknown"] <- NA  
snp_sub$Position[snp_sub$Position == "multiple"] <- NA 
View(snp_sub)
```
Removes the Sample_ID,JD_OTU, and Group from the files and transposes the data frame. After the removal, the files are viewed to verify.
```{r}
maize_rmcol <- subset(maize, select=-c(Sample_ID,JG_OTU,Group))
maize_t <- as.data.frame(t(maize_rmcol))
View(maize_t)
teosinte_rmcol <- subset(teosinte, select=-c(Sample_ID,JG_OTU,Group))
teosinte_t <- as.data.frame(t(teosinte_rmcol))
View(teosinte_t)
```
Combine the SNP File and Gentoype File for Maize and Teosinte. 
```{r}
maize_t <- data.frame(SNP_ID = row.names(maize_t), maize_t)
maize_combined <- merge(snp_sub,maize_t,by = "SNP_ID")
teosinte_t <- data.frame(SNP_ID = row.names(teosinte_t), teosinte_t)
teosinte_combined <- merge(snp_sub,teosinte_t,by = "SNP_ID")
```
Removes the markers that have unknown or multiple Genotype_Calls in the columns of chromosome or position.
```{r}
maize_filtered <- na.omit(maize_combined)
teosinte_filtered <- na.omit(teosinte_combined)
```
Order the positions in ascending order for Maize and Teosinte
```{r}
maize_filtered$Position <- as.numeric(as.character(maize_filtered$Position))
maize_incr<-maize_filtered %>% arrange(Chromosome,Position)
teosinte_filtered$Position <- as.numeric(as.character(teosinte_filtered$Position))
teosinte_incr<-teosinte_filtered %>% arrange(Chromosome,Position)
```
Replaces the ? in the data frame
```{r}
maize_incr <- as.matrix(maize_incr)
maize_incr[maize_incr=="?/?"]<-"?"
maize_incr_replaced <- as.data.frame(maize_incr)
teosinte_incr <- as.matrix(teosinte_incr)
teosinte_incr[teosinte_incr=="?/?"]<-"?"
teosinte_incr_replaced <- as.data.frame(teosinte_incr)
```
Writes files for Maize chromosomes that are increasing
```{r}
for (i in 1:10){
  maize_incr_replaced_sub=subset(maize_incr_replaced,maize_incr_replaced$Chromosome==i)
  maize_final_incr<- maize_incr_replaced_sub[order(strtoi(maize_incr_replaced_sub$Position)),]
write.table(maize_final_incr, file =sprintf("Maize_Chromosome_%i_Increasing.txt",i), sep=" ", quote=FALSE, row.names=FALSE, col.names=FALSE)}
```
Writes files for Teosinte chromosomes that are increasing
```{r}
for (i in 1:10){
  teosinte_incr_replaced_sub=subset(teosinte_incr_replaced,teosinte_incr_replaced$Chromosome==i)
  teosinte_final_incr<- teosinte_incr_replaced_sub[order(strtoi(teosinte_incr_replaced_sub$Position)),]
write.table(teosinte_final_incr, file =sprintf("Teosinte_Chromosome_%i_Increasing.txt",i), sep=" ", quote=FALSE, row.names=FALSE, col.names=FALSE)}
```
Order the positions in descending order for Maize and Teosinte
```{r}
maize_filtered$Position <- as.numeric(as.character(maize_filtered$Position))
maize_decr<-maize_filtered %>% arrange(Chromosome,Position)
teosinte_filtered$Position <- as.numeric(as.character(teosinte_filtered$Position))
teosinte_decr<-teosinte_filtered %>% arrange(Chromosome,Position)
```
Replaces the ? in the data frame
```{r}
maize_decr <- as.matrix(maize_decr)
maize_decr[maize_decr=="?/?"]<-"-"
maize_decr_replaced <- as.data.frame(maize_decr)
teosinte_decr <- as.matrix(teosinte_decr)
teosinte_decr[teosinte_decr=="?/?"]<-"-"
teosinte_decr_replaced <- as.data.frame(teosinte_decr)
```
Writes files for Maize chromosomes that are decreasing
```{r}
for (i in 1:10){
  maize_decr_replaced_sub=subset(maize_decr_replaced,maize_decr_replaced$Chromosome==i)
  maize_final_decr<- maize_decr_replaced_sub[order(strtoi(maize_decr_replaced_sub$Position)),]
write.table(maize_final_decr, file =sprintf("Maize_Chromosome_%i_Decreasing.txt",i), sep=" ", quote=FALSE, row.names=FALSE, col.names=FALSE)}
```
Writes files for Teosinte chromosomes that are decreasing
```{r}
for (i in 1:10){
  teosinte_decr_replaced_sub=subset(teosinte_decr_replaced,teosinte_decr_replaced$Chromosome==i)
  teosinte_final_decr<- teosinte_decr_replaced_sub[order(strtoi(teosinte_decr_replaced_sub$Position)),]
write.table(teosinte_final_decr, file =sprintf("Teosinte_Chromosome_%i_Decreasing.txt",i),sep=" ", quote=FALSE, row.names=FALSE, col.names=FALSE)
}
```


Part II


Removes uninformative column
```{r}
genotype_rmcol <- subset(genotype, select=-c(JG_OTU))
```
Transposes the dataframe
```{r}
genotype_t <- as.data.frame(t(genotype_rmcol))
```
Renames SNP_ID for merging in future steps.
```{r}
genotype_renamed <- data.frame(SNP_ID = row.names(genotype_t), genotype_t)
```
Selects the informative columns from the SNP file
```{r}
snp_sub_nofilter <- select(snppos,SNP_ID,Chromosome,Position)
```
Merges the dataset
```{r}
genotype_snp_combined <- merge(snp_sub_nofilter,genotype_renamed,by = "SNP_ID")
```
Melts the combined or merged dataset
```{r}
melt_maize_teosinte_snp_combined <- melt(genotype_snp_combined)
```
Plot 1A:The graph shows that ZMMLR, ZMPBA, and ZMMIL are the most important groups.
```{r}
ggplot(genotype, aes(Group)) + geom_bar()

```
Plot 1B:Plots the number of SNPs for each chromosome and plots the SNPs that have unknown and multiple positions
```{r}
ggplot(genotype_snp_combined, aes(Chromosome)) + ylab("# of SNP") + geom_bar()
```

#Missing data and amount of heterozyosity

Removes uninformative columns and recodes ?/? as NA for the missing data
```{r}
genotype_rmcol <- subset(genotype, select=-c(JG_OTU))
genotype_rmcol[ genotype_rmcol == "?/?" ] = NA
```
Melts Dataframe
```{r}
melted_genotype <- melt(genotype_rmcol,id=c("Sample_ID","Group"))
```
Adding columns and column names
```{r}
colnames(melted_genotype)[3:4] <- c("SNP_ID", "Genotype_Call")
```
Develops homozygous column for genotypes
```{r}
melted_genotype$isHomozygous <- (melted_genotype$Genotype_Call=="A/A" | melted_genotype$Genotype_Call=="C/C" | melted_genotype$Genotype_Call=="G/G" | melted_genotype$Genotype_Call=="T/T")
```
Arranges the data by Group then by Sample_ID
```{r}
melted_genotype_arr <-arrange(melted_genotype, Group, Sample_ID)
```
Sums up across all of the genotype classes
```{r}
porportion_snps <- ddply(melted_genotype_arr, c("Sample_ID"), summarize, Homozygous=sum(isHomozygous, na.rm=TRUE), Heterozygous=sum(!isHomozygous, na.rm=TRUE), Missing=sum(is.na(isHomozygous)))
```
Melts the dataframe and renames variable to Genotype_Class
```{r}
melted_porportion_snps <- melt(porportion_snps, measure.vars = c("Homozygous", "Heterozygous", "Missing"))
colnames(melted_porportion_snps)[2] <- "Genotype_Class"
```
#Plot 2A: Plots based on Sample_ID
```{r}
ggplot(melted_porportion_snps,aes(x = Sample_ID, y= value, fill=Genotype_Class)) + ylab("Proportion of SNPs") + geom_bar(stat = "identity", position = "stack")
```
Sums up across all of the genotype classes
```{r}
porportion_snps_group <- ddply(melted_genotype_arr, c("Group"), summarize, Homozygous=sum(isHomozygous, na.rm=TRUE), Heterozygous=sum(!isHomozygous, na.rm=TRUE), Missing=sum(is.na(isHomozygous)))
```
Melts the dataframe and renames variable to Genotype_Class
```{r}
melted_porportion_snps_group <- melt(porportion_snps_group, measure.vars = c("Homozygous", "Heterozygous", "Missing"))
colnames(melted_porportion_snps_group)[2] <- "Genotype_Class"
```
Plot 2B: Plots based on Group
```{r}
ggplot(melted_porportion_snps_group,aes(x = Group, y= value, fill=Genotype_Class)) + ylab("Proportion of SNPs") + geom_bar(stat = "identity", position = "stack")
```

#Visualization

Plot 3: Plots the SNPs on there respective chromosomes.
```{r}
ggplot(melt_maize_teosinte_snp_combined, aes(SNP_ID,Chromosome)) + geom_point()
```

















